### ISSUE TITLE
Request for DPO training data with reward values

### ISSUE THREAD

**Issue by pspdada** [Kwai-YuanQi/MM-RLHF#6]
Great work! I noticed that in the training data you provided (https://huggingface.co/datasets/yifanzhang114/MM-RLHF/tree/main), the dpo_pairs.jsonl file contains positive and negative sample pairs, but it does not include the reward values provided by the reward model. Would it be possible to include this field in the file to make it more convenient to use?

**Comment by yfzhang114** [comment]
We apologize for not maintaining data formats with rewards. We recommend directly using `yifanzhang114/MM-RLHF-Reward-7B-llava-ov-qwen` for generation.

**Comment by pspdada** [comment]
I noticed that in the MM-DPO Algorithm you mentioned, part of the DPO loss includes a reward margin: δ = r(yw) − r(yl). However, I didn't see this part in the code. Do you have any references for this?

Also, do you have any examples of scores provided by the reward model for your training data?

**Comment by yfzhang114** [comment]
for example 

{"id": "ai2d_azuregpt_4569_175031", "image": "long/ai2d_azuregpt_4569.png", "prompt": "Here is a diagram figure extracted from some Grade 1 - 6 science books.\nPlease first describe the content of this figure in detail, including how the knowledge visually displayed in the diagram.\nThen start with a section title \"related knowledge:\", briefly and concisely highlight the related domain knowledge and theories that underly this diagram. Note that you do not need to provide much detail. Simply cover the most important concepts.", "chosen": "This figure illustrates three different types of solar eclipses: partial eclipse, annular eclipse, and total eclipse. The diagram visually displays these phenomena using simplified representations of the sun and moon.\n\n1. Partial Eclipse: Shown on the left, this image depicts a bright orange-yellow disc (representing the sun) partially obscured by a dark shape (representing the moon), creating a crescent-like appearance.\n\n2. Annular Eclipse: The middle image shows a thin, bright ring of light surrounding a dark circular center. This represents when the moon is centered in front of the sun but doesn't completely cover it, leaving a \"ring of fire\" effect.\n\n3. Total Eclipse: On the right, we see a completely dark circular center surrounded by a bright white halo or corona. This represents the moon fully blocking the sun's disc, allowing only the sun's outer atmosphere to be visible.\n\nEach type of eclipse is clearly labeled beneath its respective image, making it easy to understand and differentiate between the three phenomena.\n\nRelated knowledge:\n\n- Solar system mechanics and orbits\n- Relative sizes and distances of the sun, moon, and Earth\n- Light and shadow principles\n- Astronomical alignments\n- Periodicity of eclipses\n- Earth's rotation and revolution\n- Lunar phases and cycles\n- Solar corona and chromosphere\n- Safe eclipse viewing practices\n- Cultural and historical significance of eclipses", "rejected": "This illustration depicts three distinct categories of solar occlusions: partial occlusion, annular occlusion, and total occlusion. The diagram visually presents these events using simplified representations of the star and satellite.\n\n1. Partial Occlusion: Shown on the left, this image depicts a bright orange-yellow sphere (representing the star) partially obscured by a dark shape (representing the satellite), creating a crescent-like appearance.\n\n2. Annular Occlusion: The middle image shows a thin, bright ring of illumination surrounding a dark circular center. This represents when the satellite is centered in front of the star but doesn't completely cover it, leaving a \"ring of fire\" effect.\n\n3. Total Occlusion: On the right, we see a completely dark circular center surrounded by a bright white halo or aura. This represents the satellite fully blocking the star's sphere, allowing only the star's outer atmosphere to be visible.\n\nEach type of occlusion is clearly labeled beneath its respective image, making it easy to understand and differentiate between the three events.\n\nRelated knowledge:\n\n- Celestial mechanics and orbits\n- Relative sizes and distances of the star, satellite, and planet\n- Light and shadow principles\n- Astronomical alignments\n- Periodicity of occlusions\n- Planet's rotation and revolution\n- Satellite phases and cycles\n- Star's corona and chromosphere\n- Safe occlusion viewing practices\n- Cultural and historical significance of occlusions", "chosen_reason": "The content in the diagram is accurately described and relatively detailed.", "rejected_reason": "Negative sample with incorrect answer.", "ls_factor": 0.10097578779650518}

**Comment by yfzhang114** [comment]
<img width="979" alt="Image" src="https://github.com/user-attachments/assets/01d19e9b-9c22-40e3-a142-25de0ee143fc" />

**Comment by pspdada** [comment]
> <img alt="Image" width="979" src="https://private-user-images.githubusercontent.com/53161950/437060797-01d19e9b-9c22-40e3-a142-25de0ee143fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU1MDUxMDksIm5iZiI6MTc0NTUwNDgwOSwicGF0aCI6Ii81MzE2MTk1MC80MzcwNjA3OTctMDFkMTllOWItOWMyMi00MGUzLWExNDItMjVkZTBlZTE0M2ZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI0VDE0MjY0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxNzBkMjkyYjQ2MGI4ZWVhNGM5ZjJlYTczZDRlYjcxOGE0NTkwZmM2ZTQyZDdlNTVkYmZiMTgyMDI3Nzg3YTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QIBYum_1MiW0Tsmpl4y4YQ-fGxGUWZDeJMv7U1L-ecs">

I have a rough understanding now. So, may I ask how you use the reward model to score the training data for DPO?

**Comment by yfzhang114** [comment]
You can use any reward model to score the data. In our dataset, each item consists of a chosen and a rejected response. You simply compute the reward for both, and then calculate the ls_factor based on the difference in their rewards.

**Comment by pspdada** [comment]
> You can use any reward model to score the data. In our dataset, each item consists of a chosen and a rejected response. You simply compute the reward for both, and then calculate the ls_factor based on the difference in their rewards.

Thanks for your reply. I noticed that in your paper, the model you proposed, MM-RLHF-Reward-7B-llava-ov-qwen, is used to score the training samples. How can I use this model to score the training samples? Is there any example code available for scoring with this reward model?

**Comment by yfzhang114** [comment]
<img width="881" alt="Image" src="https://github.com/user-attachments/assets/86f10671-76b3-4ce4-981b-acc95a4ac87f" />

**Comment by pspdada** [comment]
> <img alt="Image" width="881" src="https://private-user-images.githubusercontent.com/53161950/437375178-86f10671-76b3-4ce4-981b-acc95a4ac87f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU1ODMzMzQsIm5iZiI6MTc0NTU4MzAzNCwicGF0aCI6Ii81MzE2MTk1MC80MzczNzUxNzgtODZmMTA2NzEtNzZiMy00Y2U0LTk4MWItYWNjOTVhNGFjODdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDEyMTAzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEwM2ZjMzM1YjljODQ4NzRmNTgzNDhhZGQwODNmZDY2NDM0NWNlMzgxZWM2MmRmYzAyYzI1YzcxNDkzNWJmN2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3A1ZiN0Kcp0ts2FJvk9LXebtST2qz4bhY0773i-q-Js">

Thank you for your reply! I noticed that there are already two fields, `chosen_reason` and `rejected_reason`, in the `dpo_pairs` file. Can I use these two fields to simplify the scoring process of the reward model? That is, can I set `wo_critic` to true? What is the difference between this and letting the reward model generate the critic itself?

**Comment by yfzhang114** [comment]
=wo critic will lead to worse performance

**Comment by pspdada** [comment]
> =wo critic will lead to worse performance

I have a rough understanding now, and I misunderstood the content in the code earlier. So, what will the effect be if I use `use_gt_critic`?

**Comment by yfzhang114** [comment]
Sorry for the delayed reply. If you have a ground truth critic, then using use_gt_critic will use ground truth critic as the condition of the reward head
